### DISCUSSION TITLE
How to have all child relationship description display on parent context view?

### DISCUSSION THREAD

**Discussion by myste1tainn** [structurizr/java#300] [Q&A]
### Question

First of all, I think **structurizr** is really great, thanks for the efforts.

I'm trying to achieve certain views where relationship would be modified to be clearer for the viewer, here goes.

Considering that I have this code 

```plantuml
workspace {
    model {
        acme = softwareSystem "Acme Software" {
            aContainer = container "Container A" {
                a1 = component "A1"
                a2 = component "A2"
                a3 = component "A3"
            }
            bContainer = container "Container B" {
                b1 = component "B1"
                b2 = component "B2"
                b3 = component "B3"
            }

            a1 -> b1 "a1-to-b1 description"
            a2 -> b2 "a2-to-b2 description"
            a3 -> b3 "a3-to-b3 description"
        }
    }
    views {
        container acme "Acme" {
            include *
        }
    }
}
```

The diagram I got looks like this
![image](https://github.com/structurizr/java/assets/8848587/31657be7-1250-414c-adad-f6db52e4c1ae)
![image](https://github.com/structurizr/java/assets/8848587/9acbad2b-1fc4-4b5e-be6c-a2de22aa35c9)

But in fact, what I want to achieve is like either of this, on the container (parent level), without changing the children relations (modifying only the views)
![image](https://github.com/structurizr/java/assets/8848587/2afa1170-f90f-4fa8-ac47-37c96e88c51e)
![image](https://github.com/structurizr/java/assets/8848587/525c2aa0-7e2d-4670-91b5-e89a4230d842)

The goal is to depict and reuses all child relationship to display on the parent level so that it can be understood also on the parent level for what functionality it is trying to reaches for.

Is this possible?

If not, then another way I'm thinking is to find a way to somehow "just remove the relationship description" from the parent level. So that it wouldn't be misleading for the viewer.

Please help, thanks in advance!

PS. The other part I noticed is how, on the parent with children-to-children relationship like this, "only the 1st description" of the relationship get displayed, not sure if this is intentional or a bug? because as you can see on the parent level it becomes misleading.

**Comment by simonbrowndotje** [comment]
The following DSL will generate this view (see https://docs.structurizr.com/java/implied-relationships#implied-relationships and https://docs.structurizr.com/dsl/cookbook/implied-relationships/ for an explanation of how the implied relationships feature works):

![image](https://github.com/structurizr/java/assets/1009874/c2abbc62-0f94-4b4f-af82-919ddab68a87)

```
workspace {

    model {
        !script groovy {
            workspace.model.setImpliedRelationshipsStrategy(new com.structurizr.model.CreateImpliedRelationshipsUnlessSameRelationshipExistsStrategy());
        }

        acme = softwareSystem "Acme Software" {
            aContainer = container "Container A" {
                a1 = component "A1"
                a2 = component "A2"
                a3 = component "A3"
            }
            bContainer = container "Container B" {
                b1 = component "B1"
                b2 = component "B2"
                b3 = component "B3"
            }

            a1 -> b1 "a1-to-b1 description"
            a2 -> b2 "a2-to-b2 description"
            a3 -> b3 "a3-to-b3 description"
        }
    }
    views {
        container acme "Acme" {
            include *
        }
    }
}
```
